go/vt/sqlparser/ast.go

```diff
@@ -2372,6 +2372,15 @@ func (node *DDL) alterFormat(buf *TrackedBuffer) {
 		default:
 			buf.Myprintf(" drop constraint %s", node.TableSpec.Constraints[0].Name)
 		}
+	} else if node.ConstraintAction == RenameStr && node.TableSpec != nil && len(node.TableSpec.Constraints) == 2 {
+		switch node.TableSpec.Constraints[0].Details.(type) {
+		case *ForeignKeyDefinition:
+			buf.Myprintf(" rename constraint %s to", node.TableSpec.Constraints[0].Name)
+		}
+		switch node.TableSpec.Constraints[1].Details.(type) {
+		case *ForeignKeyDefinition:
+			buf.Myprintf(" %s", node.TableSpec.Constraints[1].Name)
+		}
```

Comment:
Do we need to switch on the type of `Details`? I see that this field will currently always be a `ForeignKeyDefinition`, but it seems like that would be likely to change in the future if we expanded this to support renaming other constraints, and then this code wouldn't print out the statement correctly anymore. Not a huge deal, since we should catch that in development, but seems like it could be simpler if we just printed out the constraint name without asserting the type.